File: c_c++/numberOf1Bits.c
#include <stdio.h>

int hammingWeight( unsigned n) {
	int ans = 0;
	while( n != 0 )
	{
		ans += 1;
		n = n & ( n -1 );
	}
	return ans;
}

int main () {

	printf( "%d", hammingWeight (11 ));
	return 0;
}

File: java/src/leetcode_in_java/NumberOf1Bits.java
package leetcode_in_java;

/**
 * Created by bopang on 15/8/1.
 */
class NumberOf1Bits {
    public int hammingWeight(int n) {
        int ans = 0;
        while (n != 0) {
            n = n & (n - 1);
            ans++;
        }
        return ans;
    }
}
